SU2_CFD/include/fluid/CFluidScalar.hpp

```diff
@@ -0,0 +1,113 @@
+  unsigned short n_species_mixture = 0;               /*!< \brief Number of species in mixture. */
+  su2double Gas_Constant = 0.0;                       /*!< \brief Specific gas constant. */
+  su2double Gamma = 0.0;                              /*!< \brief Ratio of specific heats of the gas. */
+  su2double Pressure_Thermodynamic = 0.0;             /*!< \brief Constant pressure thermodynamic. */
+
+  bool wilke;
+  bool davidson;
+
+  std::vector<su2double> massFractions;               /*!< \brief Mass fractions of all species. */
+  std::vector<su2double> moleFractions;               /*!< \brief Mole fractions of all species. */
+  std::vector<su2double> molarMasses;                 /*!< \brief Molar masses of all species. */
+  std::vector<su2double> laminarViscosity;            /*!< \brief Laminar viscosity of all species. */
+  std::vector<su2double> specificHeat;                /*!< \brief Specific heat of all species. */
+  std::vector<su2double> laminarthermalConductivity;  /*!< \brief Laminar thermal conductivity of all species. */
+
+  static const int ARRAYSIZE = 100;
+  std::unique_ptr<CViscosityModel> LaminarViscosityPointers[ARRAYSIZE];
+  std::unique_ptr<CConductivityModel> ThermalConductivityPointers[ARRAYSIZE];
+
+ public:
+  /*!
+   * \brief Constructor of the class.
+   */
+  CFluidScalar(su2double val_Cp, su2double val_gas_constant, su2double val_operating_pressure, CConfig *config);
+  /*!
+   * \brief Set viscosity model.
+   */
+  void SetLaminarViscosityModel(const CConfig* config) override;
+
+  /*!
+   * \brief Set thermal conductivity model.
+   */
+  void SetThermalConductivityModel(const CConfig* config) override;
+  
+  /*!
+   * \brief Set the Dimensionless State using Temperature.
+   * \param[in] t - Temperature value at the point.
+   */
+  void SetTDState_T(su2double t){
+    /*--- The EoS only depends upon temperature. ---*/
+    Temperature = t;
+    Density = Pressure / (Temperature * Gas_Constant);
+  }
+  /*!
+   * \brief Set the Dimensionless State using Temperature.
+   * \param[in] t - Temperature value at the point.
+   */
+
+  void SetTDState_T(const su2double val_temperature, const su2double * const val_scalars);
+
+   /*!
+   * \brief Get fluid dynamic viscosity.
+   */
+  inline su2double GetLaminarViscosity() override {
+    LaminarViscosity->SetViscosity(Temperature, Density);
+    Mu = LaminarViscosity->GetViscosity();
+    LaminarViscosity->SetDerViscosity(Temperature, Density);
+    dmudrho_T = LaminarViscosity->Getdmudrho_T();
+    dmudT_rho = LaminarViscosity->GetdmudT_rho();
+    return Mu;
+  }
+  /*!
+   * \brief Get fluid thermal conductivity.
+   */
+  inline su2double GetThermalConductivity() override { 
+    ThermalConductivity->SetConductivity(Temperature, Density, Mu, Mu_Turb, Cp);
+    Kt = ThermalConductivity->GetConductivity();
+    ThermalConductivity->SetDerConductivity(Temperature, Density, dmudrho_T, dmudT_rho, Cp);
+    dktdrho_T = ThermalConductivity->Getdktdrho_T();
```

Comment:
These functions look the same as the parent class CFluidModel, why do you need to override them?

Reply:
Yes they do not need to be override. Thanks!!